packages/analytics-js-integrations/__tests__/integrations/Amplitude/amplitude.test.js

```diff
@@ -247,11 +247,118 @@ describe('Amplitude identify tests', () => {
+      trackNamedPages: true,
+      trackCategorizedPages: true,
+      attribution: true,
+      flushQueueSize: 30,
+      flushIntervalMillis: 1000,
+      trackNewCampaigns: true,
+      trackRevenuePerProduct: false,
+      preferAnonymousIdForDeviceId: false,
+      traitsToSetOnce: ['email', 'name'],
+      traitsToIncrement: ['age'],
+      appendFieldsToEventProps: false,
+      unsetParamsReferrerOnNewSession: false,
+      trackProductsOnce: false,
+      versionName: '1.0.0',
+      groupTypeTrait: 'groupType',
+      groupValueTrait: 'groupValue',
+    };
+
+    const analytics = {
+      logLevel: 'debug',
+      getAnonymousId: () => 'ANONYMOUS_ID',
+    };
+
+    const destinationInfo = {
+      shouldApplyDeviceModeTransformation: true,
+      propagateEventsUntransformedOnError: false,
+      destinationId: 'DESTINATION_ID',
+    };
+
+    const rudderElement = {
+      message: {
+        context: {},
+        integrations: {
+          Amplitude: {
+            fieldsToUnset: ['unsetField', 'objUn.innerObj.ina'],
+          },
+        },
+      },
+    };
+    window.amplitude.identify = jest.fn();
+    const amplitude = new Amplitude(config, analytics, destinationInfo);
+    amplitude.init();
+    amplitude.identify(rudderElement);
+    expect(window.amplitude.identify.mock.calls[0][0]._q).toEqual([
+      {
+        args: ['unsetField'],
+        name: 'unset',
+      },
+      {
+        args: ['objUn.innerObj.ina'],
+        name: 'unset',
+      },
+    ]);
+  });
+  it('should identify user with no traits and empty unset field', () => {
+    const config = {
+      apiKey: 'YOUR_AMPLITUDE_API_KEY',
+      trackAllPages: true,
+      trackNamedPages: true,
+      trackCategorizedPages: true,
+      attribution: true,
+      flushQueueSize: 30,
+      flushIntervalMillis: 1000,
+      trackNewCampaigns: true,
+      trackRevenuePerProduct: false,
+      preferAnonymousIdForDeviceId: false,
+      traitsToSetOnce: ['email', 'name'],
+      traitsToIncrement: ['age'],
+      appendFieldsToEventProps: false,
+      unsetParamsReferrerOnNewSession: false,
+      trackProductsOnce: false,
+      versionName: '1.0.0',
+      groupTypeTrait: 'groupType',
+      groupValueTrait: 'groupValue',
+    };
+
+    const analytics = {
+      logLevel: 'debug',
+      getAnonymousId: () => 'ANONYMOUS_ID',
+    };
+
+    const destinationInfo = {
+      shouldApplyDeviceModeTransformation: true,
+      propagateEventsUntransformedOnError: false,
+      destinationId: 'DESTINATION_ID',
+    };
+
+    const rudderElement = {
+      message: {
+        context: {},
+        integrations: {
+          Amplitude: {
+            fieldsToUnset: [],
+          },
+        },
+      },
+    };
+    window.amplitude.identify = jest.fn();
+    const amplitude = new Amplitude(config, analytics, destinationInfo);
+    amplitude.init();
+    amplitude.identify(rudderElement);
+    expect(window.amplitude.identify).toHaveBeenCalledTimes(0);
```

Comment:
@anantjain45823 why shouldn't the identify call be triggered here ?

Reply:
Because there is nothing to be identified.
Previosuly we used to send identify call only if traits were available otherwise we just went through as it is.
But now we are checking if there is unset fields or not.
In case there is no `fieldsToUnset` and no traits we are not making the identify call to amplitude